solidity/contracts/lib/zeto_base.sol

```diff
@@ -118,4 +190,67 @@ abstract contract ZetoBase is IZetoBase, ZetoCommon {
         }
         emit UTXOMint(utxos, msg.sender, data);
     }
+    // Locks the UTXOs so that they can only be spent by submitting the appropriate
+    // proof from the Eth account designated as the "delegate". This function
+    // should be called by escrow contracts that will use uploaded proofs
+    // to execute transactions, in order to prevent the proof from being used
+    // by parties other than the escrow contract.
+    function _lock(
+        uint256[] memory inputs,
+        uint256[] memory outputs,
+        uint256[] memory lockedOutputs,
+        address delegate,
+        bytes calldata data
+    ) public {
+        for (uint256 i = 0; i < lockedOutputs.length; ++i) {
+            if (lockedOutputs[i] == 0) {
+                continue;
+            }
+            if (
+                delegates[lockedOutputs[i]] != address(0) &&
+                delegates[lockedOutputs[i]] != msg.sender
+            ) {
+                revert UTXOAlreadyLocked(lockedOutputs[i]);
+            }
+            delegates[lockedOutputs[i]] = delegate;
```

Comment:
Where do we set `_lockedUTXOs`?

Reply:
OK I see that it's in another method, `processInputsAndOutputs`. It's slightly confusing that the setup of the locked states is split up. Maybe this method just needs a different name?